salt/grains/core.py

```diff
@@ -408,60 +408,61 @@ def _aix_cpudata():
         data += '\n'
 
-    cpuarch_regexes = [
-        re.compile(r) for r in [
-            r'(?im)^\s*Processor\s+Type:\s+(\S+)',  # prtconf
+        cpuarch_regexes = [
+            re.compile(r) for r in [
+                r'(?im)^\s*Processor\s+Type:\s+(\S+)',  # prtconf
+            ]
         ]
-    ]
 
-    cpuflags_regexes = [
-        re.compile(r) for r in [
-            r'(?im)^\s*Processor\s+Version:\s+(\S+)',  # prtconf
+        cpuflags_regexes = [
+            re.compile(r) for r in [
+                r'(?im)^\s*Processor\s+Version:\s+(\S+)',  # prtconf
+            ]
         ]
-    ]
 
-    cpumodel_regexes = [
-        re.compile(r) for r in [
-            r'(?im)^\s*Processor\s+Implementation\s+Mode:\s+(.*)',  # prtconf
+        cpumodel_regexes = [
+            re.compile(r) for r in [
+                r'(?im)^\s*Processor\s+Implementation\s+Mode:\s+(.*)',  # prtconf
+            ]
         ]
-    ]
 
-    numcpus_regexes = [
-        re.compile(r) for r in [
-            r'(?im)^\s*Number\s+Of\s+Processors:\s+(\S+)',  # prtconf
+        numcpus_regexes = [
+            re.compile(r) for r in [
+                r'(?im)^\s*Number\s+Of\s+Processors:\s+(\S+)',  # prtconf
+            ]
         ]
-    ]
 
-    for regex in cpuarch_regexes:
-        res = regex.search(data)
-        if res and len(res.groups()) >= 1:
-            grains['cpuarch'] = res.group(1).strip().replace("'", "")
-            break
+        for regex in cpuarch_regexes:
+            res = regex.search(data)
+            if res and len(res.groups()) >= 1:
+                grains['cpuarch'] = res.group(1).strip().replace("'", "")
+                break
 
-    for regex in cpuflags_regexes:
-        res = regex.search(data)
-        if res and len(res.groups()) >= 1:
-            grains['cpu_flags'] = res.group(1).strip().replace("'", "")
-            break
+        for regex in cpuflags_regexes:
+            res = regex.search(data)
+            if res and len(res.groups()) >= 1:
+                grains['cpu_flags'] = res.group(1).strip().replace("'", "")
+                break
 
-    for regex in cpumodel_regexes:
-        res = regex.search(data)
-        if res and len(res.groups()) >= 1:
-            grains['cpu_model'] = res.group(1).strip().replace("'", "")
-            break
+        for regex in cpumodel_regexes:
+            res = regex.search(data)
+            if res and len(res.groups()) >= 1:
+                grains['cpu_model'] = res.group(1).strip().replace("'", "")
+                break
 
-    for regex in numcpus_regexes:
-        res = regex.search(data)
-        if res and len(res.groups()) >= 1:
-            grains['num_cpus'] = res.group(1).strip().replace("'", "")
-            break
+        for regex in numcpus_regexes:
+            res = regex.search(data)
+            if res and len(res.groups()) >= 1:
+                grains['num_cpus'] = res.group(1).strip().replace("'", "")
+                break
```

Comment:
Like I already suggested earlier, this should be flattened into one loop instead of writing a very long repeated code. The example how to do this was also there but gone after you force-pushed.

Reply:
Please flatten this.